pkg/sql/schemachanger/scplan/internal/scstage/build.go

```diff
@@ -462,7 +462,7 @@ func (sb stageBuilder) hasUnmeetableOutboundDeps(n *screl.Node) (ret bool) {
 		if !sb.isUnmetInboundDep(de) {
 			return nil
 		}
-		if de.Kind() != scgraph.SameStagePrecedence || sb.hasUnmeetableOutboundDeps(de.From()) {
+		if (de.Kind() != scgraph.SameStagePrecedence && de.Kind() != scgraph.Precedence) || sb.hasUnmeetableOutboundDeps(de.From()) {
```

Comment:
The more commentary you can add here, the better. This change as it stands LGTM, but only because we discussed it. Say something about how these edges could be satisfied in the current stage.